pkg/tests/ossm/operator/addons_disable_test.go

```diff
@@ -0,0 +1,99 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+func TestAddonsDisable(t *testing.T) {
+	test.NewTest(t).Groups(test.Full, test.Disconnected, test.ARM).Run(func(t test.TestHelper) {
+		t.Log("This test checks if the pods are removed when addons are disabled")
+		t.Log("See https://issues.redhat.com/browse/OSSM-1490")
+		meshNamespace := env.GetDefaultMeshNamespace()
+		smcpName := env.GetDefaultSMCPName()
+
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, meshNamespace)
+		})
+
+		t.LogStep("Install SMCP and wait for it to be Ready")
+		ossm.DeployControlPlane(t)
+		oc.WaitCondition(t, meshNamespace, "Kiali", "kiali", "Successful")
+
+		t.LogStep("Wait till Grafana/Prometheus/Kiali pods are ready")
+		oc.WaitPodRunning(t, pod.MatchingSelector("app=kiali", meshNamespace))
+		oc.WaitPodRunning(t, pod.MatchingSelector("app=prometheus", meshNamespace))
+		oc.WaitPodRunning(t, pod.MatchingSelector("app=grafana", meshNamespace))
+
+		t.LogStep("Disable Grafana/Kiali addons")
+		oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", `
+spec:
+  addons:
+    grafana:
+      enabled: false
+    kiali:
+      enabled: false
+`)
+		oc.WaitSMCPReady(t, meshNamespace, smcpName)
+
+		t.LogStep("Check that Grafana/Kiali pods were deleted but not Prometheus")
+		checkThatPodWasDeleted(t, meshNamespace, "grafana")
+		checkThatPodExist(t, meshNamespace, "prometheus")
+		checkThatPodWasDeleted(t, meshNamespace, "kiali")
+
+		t.LogStep("Disable also Prometheus")
+		oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", `
+spec:
+  addons:
+    prometheus:
+      enabled: false
+`)
+		oc.WaitSMCPReady(t, meshNamespace, smcpName)
+		t.LogStep("Check that all addons pods were deleted")
+		checkThatPodWasDeleted(t, meshNamespace, "kiali")
+		checkThatPodWasDeleted(t, meshNamespace, "grafana")
+		checkThatPodWasDeleted(t, meshNamespace, "prometheus")
+	})
+}
+
+func checkThatPodWasDeleted(t test.TestHelper, ns string, nameSelector string) {
+	retry.UntilSuccess(t, func(t test.TestHelper) {
+		shell.Execute(t,
+			fmt.Sprintf("oc -n %s get pod -l app=%s", ns, nameSelector),
+			assert.OutputContains("No resources found",
+				fmt.Sprintf("%s pod was deleted", nameSelector),
+				fmt.Sprintf("%s pod was not deleted in time", nameSelector)))
```

Comment:
WDYT to add something like oc.ResourceByLabelDeleted / oc.ResourceDeleted :thinking:

Reply:
Oh sorry it will be just !oc.ResourceByLabelExists as no special output about deletion here :confounded: